cmd/substreams-sink-postgres/inject_csv.go

```diff
@@ -0,0 +1,236 @@
+	theTableName := tableName
+	if err := filler.Run(ctx); err != nil {
+		return fmt.Errorf("table filler %q: %w", theTableName, err)
+	}
+	zlog.Info("table done", zap.Duration("total", time.Since(t0)))
+	return nil
+}
+
+type TableFiller struct {
+	pqSchema string
+	tblName  string
+
+	in            dstore.Store
+	startBlockNum uint64
+	stopBlockNum  uint64
+	pool          *pgxpool.Pool
+}
+
+func NewTableFiller(pool *pgxpool.Pool, pqSchema, tblName string, startBlockNum, stopBlockNum uint64, inStore dstore.Store) *TableFiller {
+	return &TableFiller{
+		tblName:       tblName,
+		pqSchema:      pqSchema,
+		pool:          pool,
+		startBlockNum: startBlockNum,
+		stopBlockNum:  stopBlockNum,
+		in:            inStore,
+	}
+}
+
+type sortedFilenames []string
+
+func (p sortedFilenames) Len() int           { return len(p) }
+func (p sortedFilenames) Less(i, j int) bool { return p[i] > p[j] }
+func (p sortedFilenames) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+func s(str string) *string {
+	return &str
+}
+
+func extractFieldsFromFirstLine(ctx context.Context, filename string, store dstore.Store) ([]string, error) {
+	fl, err := store.OpenObject(ctx, filename)
+	if err != nil {
+		return nil, fmt.Errorf("opening csv: %w", err)
+	}
+	defer fl.Close()
+
+	r := csv.NewReader(fl)
+	out, err := r.Read()
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func (t *TableFiller) Run(ctx context.Context) error {
+	zlog.Info("table filler", zap.String("table", t.tblName))
+
+	loadFiles, err := injectFilesToLoad(t.in, t.tblName, t.stopBlockNum, t.startBlockNum)
+	if err != nil {
+		return fmt.Errorf("listing files: %w", err)
+	}
+
+	if len(loadFiles) == 0 {
+		return fmt.Errorf("no file to process")
+	}
+	dbFields, err := extractFieldsFromFirstLine(ctx, loadFiles[0], t.in)
+	if err != nil {
+		return fmt.Errorf("extracting fields from first csv line: %w", err)
+	}
+
+	zlog.Info("files to load",
+		zap.String("table", t.tblName),
+		zap.Int("file_count", len(loadFiles)),
+		//		zap.Int("pruned_file_count", len(prunedFilenames)),
+	)
+	prunedFilenames := loadFiles
+
+	for _, filename := range prunedFilenames {
+		zlog.Info("opening file", zap.String("file", filename))
+
+		if err := t.injectFile(ctx, filename, dbFields); err != nil {
+			return fmt.Errorf("failed to inject file %q: %w", filename, err)
+		}
+
+		// FIXME: set up a way to mark progress
+		//
+		//if err := t.markProgress(ctx, filename, time.Now()); err != nil {
+		//	return fmt.Errorf("failed to mark progress file %q: %w", filename, err)
+		//}
+	}
+
+	return nil
+}
+
+func (t *TableFiller) injectFile(ctx context.Context, filename string, dbFields []string) error {
+	fl, err := t.in.OpenObject(ctx, filename)
+	if err != nil {
+		return fmt.Errorf("opening csv: %w", err)
+	}
+	defer fl.Close()
+
+	query := fmt.Sprintf(`COPY %s.%s ("%s") FROM STDIN WITH (FORMAT CSV, HEADER)`, t.pqSchema, t.tblName, strings.Join(dbFields, `","`))
```

Comment:
Do we need escaping here of `pqSchema`, `tblName` and `dbFiels`?

Reply:
dbFields are getting escaped within the join. I added escaping for schema and tblName